### ISSUE TITLE
Avoid negative properties and method names.

### ISSUE THREAD

**Issue by mrimvo** [godotengine/godot-proposals#3517]
### Describe the project you are working on

2D Pixel art games.

### Describe the problem or limitation you are having in your project

I often encounter negative properties or method names like

CollisionShape2D.disabled
Button.disabled
Timer.is_stopped()

It violates the "Don’t ever not avoid negative logic" design principle. It states that you always should use positive words like *enabled*, *is_running()*, to make coding easier to read and understand. Negative logic is used heavily throughout the whole Godot API.

Some positive examples that can also be found: Node2D.visible, Camera2D.current (instead of disabled), Particles.emitting, AnimationPlayer.is_playing(). I'm happy with those.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

For all the negative property names and methods, add the positive counterpart and mark the negative one as deprecated.

For Godot 4 maybe even consider removing the negative properties and methods (are there negative signals too?)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Eg. CollisionShape2D.enabled instead of CollisionShape2D.disabled

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes.

### Is there a reason why this should be core and not an add-on in the asset library?

The problem is in the Core.

**Comment by pycbouh** [comment]
> Replacing all the negative words with their positive counterparts helps to simplify coding and enhance readability.

That is only true depending on what is expected common use for the property, and also the reason why no principle should be implemented just for the sake of following it. If the most common use is a check for the negative state, then using a positive name would complicate the code and make readability worse. So it would be better to evaluate each case individually.

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot-proposals/issues/1376.

**Comment by mrimvo** [comment]
> If the most common use is a check for the negative state, then using a positive name would complicate the code and make readability worse.

I tend to disagree with this one. Imagine a Button, where the most common use is to be enabled.

I still find something like this really hard to read:

`Button.disabled = not should_be_clickable() or not is_shown()`

Compared to the positive way of saying the same:

`Button.enabled = should_be_clickable() and is_shown()`

**Comment by pycbouh** [comment]
I didn't claim that it applies to every case, on the contrary I mentioned it should be evaluated on the case by case basis. Though even for the button the logic is probably from the point of view that enabled is its default state, so you explicitly opt out of this state by checking the box. In code it may be less readable though.

But I can totally see how `Timer.is_stopped()` can be preferred. Same as `is_null()` or `is_empty()`, technically negative checks, are more useful than the other way around.
